lib/system.ml

```diff
@@ -82,7 +82,7 @@ let file_exists_respecting_case path f =
     let df = Filename.dirname f in
     (String.equal df "." || aux df)
     && exists_in_dir_respecting_case (Filename.concat path df) bf
-  in (!trust_file_cache || Sys.file_exists (Filename.concat path f)) && aux f
+  in (not !trust_file_cache || Sys.file_exists (Filename.concat path f)) && aux f
```

Comment:
This doesn't look right, it `trust = true`, we should not `stat` the file.

Reply:
Behaviour from 13f014ba37e0af547d57854df8926d633f9ccb7b